tests/test_interpreter.py

```diff
@@ -42,6 +43,28 @@ def test_reset():
     assert interpreter.messages == []
 
 
+def test_token_counter():
```

Comment:
Added some basic tests to make sure the logic was sound.

We could also do some actual conversations with the model and then make sure that the tokens increment as expected when using the updated `interpreter.messages` array.